libs/vkd3d/memory.c

```diff
@@ -189,7 +189,8 @@ static HRESULT vkd3d_try_allocate_device_memory(struct d3d12_device *device,
     /* buffer_mask / sampled_mask etc will generally take care of this,
      * but for certain fallback scenarios where we select other memory
      * types, we need to mask here as well. */
-    type_mask &= device->memory_info.global_mask;
+    if (!ignore_hvv_restrictions)
+        type_mask &= device->memory_info.global_mask;
```

Comment:
Seeing this now, I think the bool here is pretty goofy and we should try to avoid it.

Thinking a bit more, I'm pondering if the global_mask is even meaningful here. Somehow I don't think it's actually useful here. I'll think a bit about if we can get rid of the global mask entirely. That would allow us to just use different memory_properties and everything will work fine.